### PR TITLE
Removing more XML reading

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by alfoa** [comment]
@joshua-cogliati-inl is it ready to be reviewed?

**Comment by joshua-cogliati-inl** [comment]
Yes, it is ready to be reviewed.

File: framework/Assembler.py [hunk 1]
```diff
@@ -122,58 +128,40 @@ def _readMoreXML(self, xmlNode):
...
-                                         g=testObjects[toObjectName]))
-    if '_localReadMoreXML' in dir(self):
+      for i,token in enumerate(self._requiredAsmbObject[1]):
+        quantity = self._requiredAsmbObject[2][i]
+        if not InputData.checkQuantity(quantity, testObjects[token]):
+          self.raiseAnError(IOError, 'the object '+token+' has wrong quantity Expected: '+str(quantity)+' Found: '+str(testObjects[token])+ ' in block '+self.name)
+
+    if '_handleInput' in dir(self) and self._handleInput.__func__.__qualname__.split(".")[0] == self.__class__.__name__:
+      #_handleInput in class and not from superclass
+      print(self, self.getInputSpecification, self.getInputSpecification.__func__.__qualname__, self._handleInput, self._handleInput.__func__.__qualname__)
```
Comment (PaulTalbot-INL) [inline framework/Assembler.py:148]: Is this print statement meant to be retained?

File: framework/Assembler.py [hunk 2]
```diff
@@ -43,7 +43,12 @@ def __init__(self):
     self.name = self.__class__.__name__  # name
     if not hasattr(self, 'assemblerObjects'): # protect against polyinheritance woes
       self.assemblerObjects = {}           # {MainClassName(e.g.Distributions):[class(e.g.Models),type(e.g.ROM),objectName]}
-      self.requiredAssObject = [False, ([], [])]
+      # _requiredAsmbObject = [check_number, [name_list], [number_list]]
+      #  where name_list is the tokens required (if check_number is True)
+      #  and number_list is a string list of either the number required
+      #   - number required (if optional) or 'n' if one or more required
+      #   or '-n' if not required.
```
Comment (PaulTalbot-INL) [inline framework/Assembler.py:50]: is this description accurate with the InputData.Quantity usage?

**Comment by PaulTalbot-INL** [comment]
Is this meant to close #1290 or is there more to do for that issue?

**Comment by joshua-cogliati-inl** [comment]
> Is this meant to close #1290 or is there more to do for that issue?

I think #1290 will also require work on generateAllAssemblers in Simulation.py.

**Comment by PaulTalbot-INL** [comment]
approved to merge when tests pass.

File: framework/Assembler.py (hunk 1 above)
Comment (joshua-cogliati-inl) [reply framework/Assembler.py]: Commented out (useful for debugging but does not need to be printed most times)

File: framework/Assembler.py (hunk 2 above)
Comment (joshua-cogliati-inl) [reply framework/Assembler.py]: Fixed.  Thanks.

**Comment by PaulTalbot-INL** [comment]
From RAVEN webhooks:
``` 
a08ab100-0199-11eb-923f-a738146778fa…2020-09-28 08:48:21
Request
We couldn’t deliver this payload: timed out
```

**Pull request merged**